query-engine/core/src/interpreter/query_interpreters/nested_read.rs

```diff
@@ -190,6 +189,15 @@ pub async fn one2m(
         return Ok((ManyRecords::empty(selected_fields), None));
     }
 
+    // If we're fetching related records from a single parent, then we can apply normal pagination.
+    // If we're fetching related records from multiple parents though, we can't just apply a LIMIT/OFFSET,
+    // as we need N related records PER parent. The only known viable solution is the use of `ROW_NUMBER()`.
```

Comment:
It can be interpreted from this comment that ROW_NUMBER is actually a solution being applied, when it's not, it's something that might be applied in the future. 

```suggestion
    // If we're fetching related records from a single parent, then we can apply LIMIT/OFFSET pagination, and not 
    // require to use an in-memory processor.
    // If we're fetching related records from multiple parents though, we can't just apply that kind of pagination
    // as we need N related records PER parent. We might be able to come up with an alternative solution if we
    // used `ROW_NUMBER()` to identify the position of records in a result set.
```

Reply:
Good point! Thanks for the suggestion, I've rephrased it my way in a follow-up commit. I'll merge now 👍

Reply:
Thank you!